cognite/client/credentials.py

```diff
@@ -83,13 +83,11 @@ def __setstate__(self, state: dict[str, Any]) -> None:
 
     @abstractmethod
     def _refresh_access_token(self) -> tuple[str, float]:
-        """This method should return the access_token and expiry time"""
+        """This method should return the access_token and time until expiration (expire_in)"""
         raise NotImplementedError
 
     def __should_refresh_token(self, token: str | None, expires_at: float | None) -> bool:
-        no_token = token is None
-        token_is_expired = expires_at is None or time.time() > expires_at - self.token_expiry_leeway_seconds
-        return no_token or token_is_expired
+        return token is None or expires_at is None or expires_at > time.time() + self.token_expiry_leeway_seconds
```

Comment:
Hmm, doesn't this change behaviour?

`time > expiry - leeway` !== `exp > time + leeway`

Also I prefer keeping the separate parts of the conditional as variables.